### PR TITLE
support merge

### PR DESCRIPTION
**Pull request by BusyJay**
This pr adds protocols using in merge.

### CODE REVIEW

**Comment by disksing** [comment]
Is this branch ready to merge now?

File: proto/raft_cmdpb.proto [hunk 1]
```diff
@@ -133,6 +133,27 @@ message VerifyHashRequest {
 
 message VerifyHashResponse {}
 
+message PreMergeRequest {
```
[the remark this answers is not in the document]
Comment (BusyJay) [reply proto/raft_cmdpb.proto]: I'm OK with `PrepareMergeRequest` and `CommitMergeRequest`. But I'm worried about `RollbackMergeRequest` may indicates that `Merge` can be reverted.

[the remark this answers is not in the document]
Comment (overvenus) [reply proto/raft_cmdpb.proto]: I think rollback is different from revert.

**Comment by BusyJay** [comment]
PTAL

**Pull request merged**